lib/charms/prometheus_k8s/v0/prometheus.py

```diff
@@ -494,14 +494,14 @@ def _labeled_static_job_config(self, job, job_name_prefix, hosts, scrape_metadat
                     unitless_targets, labels, scrape_metadata
                 )
                 config["static_configs"].append(unitless_config)
-            else:
-                relabel_config["source_labels"].append("juju_unit")
 
             for host_name, host_address in hosts.items():
                 static_config = self._labeled_unit_config(
                     host_name, host_address, ports, labels, scrape_metadata
                 )
                 config["static_configs"].append(static_config)
+                if "juju_unit" not in relabel_config["source_labels"]:
```

Comment:
Not all scrape targets have unit labels. In particular "fully qualified hosts" (i.e. those whose address is not automatically gathered by the Prometheus charm library but set explicitly by consumer charms) do not have unit labels. Doesn't this line mean unit labels are attached to all units regardless. If so how does the unit get determined ?

Reply:
On further thought this should be fine since the `hosts` list is the automatically gathered list.